static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -313,6 +313,11 @@ function WidgetViewerModal(props: Props) {
     };
     getDiscoverTotals();
   }, [selectedQueryIndex]);
+  // State update causes browser to lose focus on the modal portal.
```

Comment:
Hmm, it shouldn't be possible to lose focus of the modal because of the focus trap stuff we have here

https://github.com/getsentry/sentry/blob/3e89d917f1cf2b84b412b45ad1a3b15da46eab71/static/app/components/globalModal/index.tsx#L151

Can you investigate if there's something wrong with how we're trapping focus?

Reply:
Seems like the focus-trap isn't catching all scenarios and the focus is leaking out on certain state updates/rerenders. I looked into the lib a bit and tried some different options but couldn't find anything that would fix this focus issue.

Updated this pr instead to attach the event listener to `document` instead of just the modal portal. This should universally solve all issues where pressing Escape sometimes doesn't close the modal.